pkg/controllers/rollout/controller.go

```diff
@@ -812,6 +821,38 @@ func handleResourceBinding(binding client.Object, q workqueue.RateLimitingInterf
 	})
 }
 
+// handleResourceBindingUpdated determines the action to take when a resource binding is updated.
+// we only care about the Available and DiffReported condition change.
+func handleResourceBindingUpdated(objectOld, objectNew client.Object, q workqueue.RateLimitingInterface) {
+	// Check if the update event is valid.
+	if objectOld == nil || objectNew == nil {
+		klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("update event is nil")), "Failed to process update event")
+		return
+	}
+	oldBinding, oldOk := objectOld.(*fleetv1beta1.ClusterResourceBinding)
+	newBinding, newOk := objectNew.(*fleetv1beta1.ClusterResourceBinding)
+	if !oldOk || !newOk {
+		klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("failed to cast runtime objects in update event to cluster resource binding objects")), "Failed to process update event")
+	}
+	if oldBinding.GetResourceVersion() != newBinding.GetResourceVersion() {
+		klog.V(2).InfoS("The binding spec have changed, need to notify rollout controller", "binding", klog.KObj(newBinding))
```

Comment:
Hi Ryan! Just a nit: as resource version would change even upon status refreshes, the log part might need to be revised.